bigchaindb/backend/changefeed.py

```diff
@@ -1 +1,89 @@
 """Changefeed interfaces for backends."""
+
+from functools import singledispatch
+
+from multipipes import Node
+
+import bigchaindb
+
+
+class ChangeFeed(Node):
+    """Create a new changefeed.
+
+    It extends :class:`multipipes.Node` to make it pluggable in other
+    Pipelines instances, and makes usage of ``self.outqueue`` to output
+    the data.
+
+    A changefeed is a real time feed on inserts, updates, and deletes, and
+    is volatile. This class is a helper to create changefeeds. Moreover,
+    it provides a way to specify a ``prefeed`` of iterable data to output
+    before the actual changefeed.
+    """
+
+    INSERT = 1
+    DELETE = 2
+    UPDATE = 4
+
+    def __init__(self, table, operation, *, prefeed=None, connection=None):
+        """Create a new ChangeFeed.
+
+        Args:
+            table (str): name of the table to listen to for changes.
+            operation (int): can be ChangeFeed.INSERT, ChangeFeed.DELETE, or
+                ChangeFeed.UPDATE. Combining multiple operation is possible
+                with the bitwise ``|`` operator
+                (e.g. ``ChangeFeed.INSERT | ChangeFeed.UPDATE``)
+            prefeed (:class:`~collections.abc.Iterable`): whatever set of data
+                you want to be published first.
+            connection (:class:`~bigchaindb.backend.connection.Connection`):
+                A connection to the database (can be None).
```

Comment:
It's not something we follow consistently, but I think a nice way of noting an argument as optional is to tag it in the object description (as seen in the [napoleon docs](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google)):

```python
"""
...
connection (:class:`~bigchaindb.backend.connection.Connection`, optional):
    A connection to the database.
"""
```

We could also add that the `connection` defaults to a new `Connection` instance.